src/Bicep.Core/Emit/TemplateEmitter.cs

```diff
@@ -38,7 +42,9 @@ public TemplateEmitter(SemanticModel model, EmitterSettings settings)
                 Formatting = Formatting.Indented
             };
 
-            new TemplateWriter(this.model, this.settings).Write(writer);
+            var emitter = new TemplateWriter(this.model, this.settings);
```

Comment:
It looks like `model` and `settings` don't change.  Can `emitter` be made into another private readonly field and reused across these methods?

Reply:
I tried this and it caused some test failures in LoadFunctionsTests -- some type changed, and the reason is not immediately evident. This suggestion does change the actual behavior though as there's a single instance of the TemplateWriter in this new case, as opposed to a new instance for each Emit() call in the original case. The single instance I figure is the cause of the test failures in some way. So leaving this as is for now.